src/sonic-config-engine/minigraph.py

```diff
@@ -47,6 +47,26 @@ def default(self, obj):
             return str(obj)
         return json.JSONEncoder.default(self, obj)
 
+def parse_device(device):
+    lo_prefix = None
+    mgmt_prefix = None
+    # don't shadow type()
+    d_type = None
+    hwsku = None
+    if str(QName(ns3, "type")) in device.attrib:
+        d_type = device.attrib[str(QName(ns3, "type"))]
+
+    for node in device:
+        if node.tag == str(QName(ns, "Address")):
+            lo_prefix = node.find(str(QName(ns2, "IPPrefix"))).text
+        elif node.tag == str(QName(ns, "ManagementAddress")):
+            mgmt_prefix = node.find(str(QName(ns2, "IPPrefix"))).text
+        elif node.tag == str(QName(ns, "Hostname")):
+            name = node.text
+        elif node.tag == str(QName(ns, "HwSku")):
+            hwsku = node.text
+    return (lo_prefix, mgmt_prefix, name, hwsku, d_type)
```

Comment:
Fixed.

---
In reply to: [125150495](https://github.com/Azure/sonic-buildimage/pull/775#discussion_r125150495) [](ancestors = 125150495)